### DISCUSSION TITLE
pass in pre-grouped (+aggregated) data

### DISCUSSION THREAD

**Discussion by githorse** [KevinVandy/material-react-table#532] [Ideas]
In some cases I might want to forgo some of MRT's dynamic features in order to dramatically improve performance (10x+) by doing my own aggregations outside of the grid (see [aggregation benchmark](https://github.com/KevinVandy/material-react-table/discussions/531)). Is there any way that might be achieved? Say I calculate my own [grouped and summed data](https://codesandbox.io/p/sandbox/8t1722?file=%2Fsrc%2FMetal.tsx%3A14%2C4-35%2C5), like this:

```
type GroupedSummedData<R> = {
  [key in string]: [
    number, // the aggregated value
    R[] // records with this key
  ]
}

const data: GroupedSummedData = {
  'Georgia': [1900, <Georgia records>],
  'Florida': [4100, <Florida records>],
  'Connecticut: [5000, <Connecticut records>]
}
```

Obviously the real data might be more complicated than this, but you get the idea. Probably this really boils down to "Can I pass in *grouped* data," because the aggregated value can just be thought of as another column value, though the Typescript is going to get hairy.

Is there some way this grouped data could be passed into the grid so that I can still use the beautiful UI and a subset of features like column drag-n-drop, column pinning, etc. while enjoying the performance benefits of a custom aggregation calculation? (I can [filter, paginate, and sort outside of the grid](https://www.material-react-table.com/docs/examples/remote#remote-data-fetching-example), but I don't see any way to group and aggregate.)

**Comment by githorse** [comment]
Note that server-side row grouping seems to be possible in [AG Grid](https://www.ag-grid.com/react-data-grid/server-side-model-grouping/#server-side-row-grouping](https://www.ag-grid.com/react-data-grid)), albeit by going through with a somewhat complicated [remote data source implementation](https://www.ag-grid.com/react-data-grid/server-side-model-grouping/#server-side-row-grouping).

**Comment by KevinVandy** [comment]
Yes, it's possible, but in TanStack Table, but I haven't done it myself or made an example yet.

**Reply by githorse** [reply]
Ok, I asked [at Tanstack](https://github.com/TanStack/table/discussions/4863) -- I'll keep trying but maybe somebody knows how to make that work.
